hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -674,15 +679,66 @@ public boolean isNearCacheEnabled() {
         return nearCacheConfig != null;
     }
 
+    /**
+     *
+     * @return
+     * @deprecated user {@link #getQueryCacheConfigs()} instead.
+     */
     public boolean isOptimizeQueries() {
-        return optimizeQueries;
+        return cacheDeserializedValues == CacheDeserializedValues.ALWAYS;
     }
 
+    /**
+     *
+     * Enable de-serialized value caching when evaluating predicates.
```

Comment:
Ca you also add to javaDoc here that what will be the behaviour when InMemoryFormat is OBJECT? Or is this only valid for InMemoryFormat#BINARY ?